students/TimPauley/session01/test_unit.py

```diff
@@ -0,0 +1,333 @@
+    def test_generate_menu(self):
+        self.assertEqual(generate_menu("Z"), False)
+        self.assertEqual(generate_menu("1").__name__, "add_new_item")
+        self.assertEqual(generate_menu("2").__name__, "item_info")
+        self.assertEqual(generate_menu("q").__name__, "exit_program")
+
+    def test_get_price(self):
+        self.assertEqual(get_price(), get_latest_price())
+
+    def test_return_item_info(self):
+        item_dict = {}
+        item_dict["item_code"] = 2
+        item_dict["item_description"] = "b"
+        item_dict["item_rental_price"] = 2.0
+        item_dict["item_price"] = 2.0
+        i1 = Inventory(**item_dict)
+        item_dict1 = {}
+        item_dict1["item_code"] = 2
+        item_dict1["item_description"] = "b"
+        item_dict1["item_rental_price"] = 2.0
+        item_dict1["item_price"] = 2.0
+        i2 = Inventory(**item_dict1)
+        f_inventory = {}
+        f_inventory["1"] = i1.return_as_dictionary()
+        f_inventory["2"] = i2.return_as_dictionary()
+        self.assertEqual(
+            return_item_info(f_inventory, "2"),
+            "product_code:1\ndescription:a\nmarket_price:1.0\nrental_price:1.0\n"
+        )
+        self.assertEqual(
+            return_item_info(f_inventory, "2"),
+            "product_code:2\ndescription:b\nmarket_price:2.0\nrental_price:2.0\n"
+        )
+        self.assertEqual(
+            return_item_info(f_inventory, "z"), "Item not found in inventory")
+
+    def test_add_new_item_furniture(self):
+        text_trap = io.StringIO()
+        sys.stdout = text_trap
+        item_dict = {}
+        item_dict["item_code"] = "1"
+        item_dict["item_description"] = "a"
+        item_dict["item_rental_price"] = "1.0"
+        item_dict["item_price"] = "1.0"
+        item_dict["item_material"] = "a"
+        item_dict["size"] = "S"
+        item_dict["style"] = "furniture"
+        user_input = ["1", "a", "1.0", "y", "a", "S"]
+        f_inventory = {}
+        f = Furniture(**item_dict)
+        f_inventory = {}
+        with patch('builtins.input', side_effect=user_input):
+            f_inventory = add_new_item(f_inventory)
+        new_dict = f_inventory["1"]
+        f = Furniture(**item_dict)
+        new_dict1 = f.return_as_dictionary()
+        self.assertEqual(new_dict["product_code"], new_dict1["product_code"])
+        self.assertEqual(new_dict["description"], new_dict1["description"])
+        self.assertEqual(new_dict["rental_price"], new_dict1["rental_price"])
+        self.assertEqual(new_dict["material"], new_dict1["material"])
+        self.assertEqual(new_dict["size"], new_dict1["size"])
+        sys.stdout = sys.__stdout__
+
+    def test_add_new_item_electrical(self):
+        text_trap = io.StringIO()
+        sys.stdout = text_trap
+        user_input = ["2", "b", "2.0", "n", "y", "b", "2.0"]
+        f_inventory = {}
+        with patch('builtins.input', side_effect=user_input):
+            f_inventory = add_new_item(f_inventory)
```

Comment:
you will have to adapt add_new_item to return something as well, for this to work.